### PR TITLE
feat: add Brotli compressor

### PR DESCRIPTION
**Pull request by ice-brontes**
## Description
Add Brotli compressor/decompressor to the project.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

**Comment by ice-brontes** [comment]
```
File extension:         .txt
Original file size:     3.09 KB
Compressed file size:   1.18 KB
Decompressed file size: 3.09 KB
Are decompressed and original files equals? - true
Compressing finished in 39 milliseconds
Decompressing finished in 39 milliseconds

File extension:         .xlsx
Original file size:     15.97 KB
Compressed file size:   13.48 KB
Decompressed file size: 15.97 KB
Are decompressed and original files equals? - true
Compressing finished in 35 milliseconds
Decompressing finished in 35 milliseconds

File extension:         .html
Original file size:     30.66 KB
Compressed file size:   4.08 KB
Decompressed file size: 30.66 KB
Are decompressed and original files equals? - true
Compressing finished in 36 milliseconds
Decompressing finished in 36 milliseconds

File extension:         .pdf
Original file size:     18.37 KB
Compressed file size:   15.23 KB
Decompressed file size: 18.37 KB
Are decompressed and original files equals? - true
Compressing finished in 43 milliseconds
Decompressing finished in 43 milliseconds

File extension:         .jpg
Original file size:     28.86 KB
Compressed file size:   28.76 KB
Decompressed file size: 28.86 KB
Are decompressed and original files equals? - true
Compressing finished in 95 milliseconds
Decompressing finished in 95 milliseconds

File extension:         .json
Original file size:     250.56 KB
Compressed file size:   7.92 KB
Decompressed file size: 250.56 KB
Are decompressed and original files equals? - true
Compressing finished in 73 milliseconds
Decompressing finished in 73 milliseconds

File extension:         .json
Original file size:     501.12 KB
Compressed file size:   7.93 KB
Decompressed file size: 501.12 KB
Are decompressed and original files equals? - true
Compressing finished in 84 milliseconds
Decompressing finished in 84 milliseconds

File extension:         .json
Original file size:     1002.24 KB
Compressed file size:   7.95 KB
Decompressed file size: 1002.24 KB
Are decompressed and original files equals? - true
Compressing finished in 561 milliseconds
Decompressing finished in 561 milliseconds

File extension:         .json
Original file size:     4.89 MB
Compressed file size:   7.97 KB
Decompressed file size: 4.89 MB
Are decompressed and original files equals? - true
Compressing finished in 634 milliseconds
Decompressing finished in 634 milliseconds

File extension:         .wav
Original file size:     413.44 KB
Compressed file size:   247.74 KB
Decompressed file size: 413.44 KB
Are decompressed and original files equals? - true
Compressing finished in 994 milliseconds
Decompressing finished in 994 milliseconds

File extension:         .mp4
Original file size:     181.78 MB
Compressed file size:   181.14 MB
Decompressed file size: 181.78 MB
Are decompressed and original files equals? - true
Compressing finished in 3 minutes 28 seconds 844 milliseconds
Decompressing finished in 3 minutes 28 seconds 844 milliseconds
```

File: lib/app/services/compressor/compress_service.dart [hunk 1]
```diff
@@ -214,6 +217,71 @@ class MediaCompressionService {
...
+      );
+      final inputData = await inputFile.readAsBytes();
+      final compressedData = brotliCompressor.encode(inputData);
+
+      // TODO: Some decompressors requires base64 converted output file
+      // Try to convert the output with base64.encode() if the backend requires it
+      return _saveBytesIntoFile(bytes: compressedData, extension: 'br');
+    } catch (error, stackTrace) {
+      Logger.log('Error during Brotli compression!', error: error, stackTrace: stackTrace);
+      throw Exception('Failed to compress file with Brotli.');
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/services/compressor/compress_service.dart]: Replaced all `Exception`s in the `CompressionService` with the custom ones.

File: lib/app/services/compressor/compress_service.dart [hunk 2]
```diff
@@ -214,6 +217,71 @@ class MediaCompressionService {
...
+  Future<File> decompressBrotli(File compressedFile, {String outputExtension = 'txt'}) async {
+    try {
+      final compressedData = await compressedFile.readAsBytes();
+      final decompressedData = brotli.decode(
+        Uint8List.fromList(compressedData),
+      );
+      return _saveBytesIntoFile(bytes: decompressedData, extension: outputExtension);
+    } catch (error, stackTrace) {
+      Logger.log('Error during Brotli decompression!', error: error, stackTrace: stackTrace);
+      throw Exception('Failed to decompress Brotli file.');
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/services/compressor/compress_service.dart]: Done.

File: lib/app/services/compressor/compress_service.dart [hunk 3]
```diff
@@ -214,6 +217,71 @@ class MediaCompressionService {
     }
   }
 
+  /// Compresses a file using the Brotli algorithm.
+  ///
+  /// This method reads the input file, compresses its content using the Brotli codec,
+  /// and writes the compressed data to a new file with a `.br` extension.
+  ///
+  /// If any errors occur during compression, they are logged, and an exception is thrown.
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/services/compressor/compress_service.dart]: Removed.

File: lib/app/services/compressor/compress_service.dart [hunk 4]
```diff
@@ -214,6 +217,71 @@ class MediaCompressionService {
...
+  /// Throws:
+  /// - [Exception] if compression fails due to invalid input or other errors.
+  ///
+  Future<File> compressWithBrotli(File inputFile) async {
+    try {
+      // The input buffer needs to be larger than 50% of the total file size
+      // for it to be compressed correctly. This is due to this issue:
+      // https://github.com/instantiations/es_compression/issues/53
+      // Once it is fixed, the current solution needs to be reconsidered.
+      const bufferPercentage = 0.6;
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/services/compressor/compress_service.dart:229]: The largest file size used during testing was just under 200 Mb.

File: lib/app/services/compressor/compress_service.dart [hunk 5]
```diff
@@ -214,6 +217,71 @@ class MediaCompressionService {
...
+      final fileSize = inputFile.lengthSync();
+
+      final brotliCompressor = BrotliCodec(
+        inputBufferLength: (fileSize * bufferPercentage).ceil(),
+      );
+      final inputData = await inputFile.readAsBytes();
+      final compressedData = brotliCompressor.encode(inputData);
+
+      // TODO: Some decompressors requires base64 converted output file
+      // Try to convert the output with base64.encode() if the backend requires it
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/services/compressor/compress_service.dart:239]: Done, removed todo

**Review by ice-alcides — APPROVED** [review]

**Review by ice-ajax — APPROVED** [review]

**Pull request merged**